### PR TITLE
atari400.cpp: Add 2-port version of a5200; fix ANTIC memory size

### PR DESCRIPTION
**Pull request by einstein95**
New working clones:
Atari 5200 (2-port) (1983, Atari)

### CODE REVIEW

**Comment by einstein95** [comment]
Part numbers taken from [the 5200 service manual](https://archive.org/details/CX_5200_Field_Service_Manual_Rev_4_July_1983_U).

**Comment by happppp** [comment]
0, not o. Also, the 2-port version can have either.

4-port version and older 2-port version: C019156-05 (there's also C019156-03 but I don't know if it's any different)
2-port version, newer revision: C019156A-08

**Comment by DavidHaywood** [comment]
if the machines were physically different, shouldn't they be treated as different machines, not a BIOS option?

**Comment by einstein95** [comment]
> 0, not o. Also, the 2-port version can have either.
> 
> 4-port version and older 2-port version: C019156-05 (there's also C019156-03 but I don't know if it's any different) 2-port version, newer revision: C019156A-08

I used `co` as that's what the other drivers in the same file are using. I'll update the name for the 2-port to be `C019156A`.

> if the machines were physically different, shouldn't they be treated as different machines, not a BIOS option?

I'm debating whether to do this as using a 4-port BIOS in a 2-port 5200 is a common modification due to the compatibility with these 3 games.

**Comment by einstein95** [comment]
I've added a clone of a5200 with player 3/4 ports removed and both BIOSes (as 2-port with 4-port BIOS was a common mod but not the other way around). Other updates are to comments in the source, removing the "sio" device (so that it no longer appears when doing `mame a5200 -listslots`), and resizing the `m_antic` memory region to be the proper size.

**Comment by cuavas** [comment]
Please give this PR a terse but more descriptive title.

**Comment by einstein95** [comment]
@cuavas Like this?

File: src/mame/drivers/atari400.cpp [hunk 1]
```diff
@@ -2344,9 +2370,11 @@ void a400_state::a5200(machine_config &config)
...
 	m_antic->set_gtia_tag(m_gtia);
 
-	m_pia->readpa_handler().set_constant(0); // FIXME: is there anything connected here
-	m_pia->readpb_handler().set_constant(0); // FIXME: is there anything connected here
-	m_pia->cb2_handler().set_nop(); // FIXME: is there anything connected here
+	// PIA chip is not present
+	m_pia->readpa_handler().set_constant(0);
+	m_pia->readpb_handler().set_constant(0);
+	m_pia->cb2_handler().set_nop();
+	config.device_remove("sio");
```
Comment (cuavas) [inline src/mame/drivers/atari400.cpp:2377]: If the PIA and SIO really aren’t present, you’re better off not adding them in the base machine configuration function.  Using `device_remove` and having vestigial devices hanging around makes life harder.

Can you make it so these things are only added for systems that actually have them present?

**Pull request merged**